lib/utils/cli.go

```diff
@@ -77,19 +76,33 @@ func InitLogger(purpose LoggingPurpose, level log.Level, verbose ...bool) {
 
 // InitLoggerForTests initializes the standard logger for tests.
 func InitLoggerForTests() {
```

Comment:
I wonder if we should get rid of `InitLoggerForTests` completely.

I see two better options:

1. We already have `InitLogger(purpose, ...)`, so can we just add a new `LoggingForTests` purpose and use that instead?
2. We have a `testlog` package under `lib/utils` - this probably makes more sense there than in a file called `cli.go`.

I would lean towards option 2 since this is test-specific code.

Reply:
Good idea. We should have one way to define logging. 

There is also `NewLoggerForTests()` in `cli.go` that is used in many places but creates a new root logger rather than inherit from the existing root, which makes it difficult to disable or set levels for logging in all tests. I've run into this multiple times when trying to disable logging.

It would be nice to have a single way the system defines a root logger, regardless of whether it is created in a `main/Run` or a test init. Subsequent loggers, such as those created by `testlog.FailureOnly` or `utils.NewLoggerForTests`, should either create a child logger from the root logger or override the root logger, so the root logger's settings are inherited.

I think we can do a combination of both 1 and 2 because `testlog.FailureOnly` calls `utils.NewLoggerForTests`. And `NewLoggerForTests` simply creates a second root logger with debug level logging. `FailureOnly` should only need to call `SetOutput` for the root logger to implement its functionality.

Suggestions:
- Importing `testlog` should init the root logger for tests by calling `InitLogger` with a new `LoggingForTests` purpose
- `NewLoggerForTests` should be removed; callers should import `testlog` and use or override the root logger
- `FailureOnly` should call `SetOutput` on the root logger
- Remove `InitLoggerForTests`